### PR TITLE
Rename `Brave Today` to `Brave News`

### PR DESCRIPTION
**Pull request by fallaciousreasoning**
Resolves https://github.com/brave/brave-browser/issues/28575

This has been on the todo list for a while. Brave Today was renamed to Brave News but we hadn't renamed the files/folders and a bunch of the internal strings.

### CODE REVIEW

**Review by fmarier — APPROVED** [review]
`browser/net/brave_network_audit_allowed_lists.h` looks good

File: components/brave_news/browser/suggestions_controller.h [hunk 1]
```diff
@@ -3,8 +3,8 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this file,
 // you can obtain one at http://mozilla.org/MPL/2.0/.
 
-#ifndef BRAVE_COMPONENTS_BRAVE_TODAY_BROWSER_SUGGESTIONS_CONTROLLER_H_
-#define BRAVE_COMPONENTS_BRAVE_TODAY_BROWSER_SUGGESTIONS_CONTROLLER_H_
+#ifndef BRAVE_COMPONENTS_brave_news_BROWSER_SUGGESTIONS_CONTROLLER_H_
```
Comment (tmancey) [inline components/brave_news/browser/suggestions_controller.h:6]: Should be uppercase, see by from "npm run lint"

File: components/brave_news/browser/test/BUILD.gn [hunk 2]
```diff
@@ -0,0 +1,43 @@
+# Copyright (c) 2021 The Brave Authors. All rights reserved.
```
Comment (tmancey) [inline components/brave_news/browser/test/BUILD.gn:1]: 2023

**Review by tmancey — APPROVED** [review]
LGTM other than linter

**Review by rillian — APPROVED** [review]
rust-deps++

Also, hooray! :)

File: components/brave_news/browser/test/BUILD.gn (hunk 2 above)
Comment (fallaciousreasoning) [reply components/brave_news/browser/test/BUILD.gn:1]: I don't think this should actually be changed, as the file has been moved

File: components/brave_news/common/pref_names.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
+// Copyright (c) 2021 The Brave Authors. All rights reserved.
```
Comment (petemill) [inline components/brave_news/common/pref_names.h:1]: I wonder why this is not treated as a rename? Any idea how we can improve it, e.g. via rename then modify? And then using a merge commit for this PR?

**Review by petemill — APPROVED** [review]
Looks good apart from some apparent git loss of history if github's UI is to be trusted. But it's minimal so that's ok if it can't be improved. Also I would have thought there'd be more Android import paths to rename, but I suppose it doesn't use import paths and the namespace was already correct.

Thanks!

File: components/brave_news/common/pref_names.h (hunk 3 above)
Comment (fallaciousreasoning) [reply components/brave_news/common/pref_names.h:1]: I guess the file contents changed too much for Git to deal with :/ We could do the merge but honestly, I don't know if it's worth it - git is a bit hopeless with renames and it should be fairly obvious what's going on from the commit name.

If this is breaking CI I think I'd be happier just updating the copyright header, tbh

Comment (petemill) [reply components/brave_news/common/pref_names.h:1]: It's less what's going on from the commit name and more when someone wants to figure out the history of the pref names and the trail stops at this commit. But it's ok if it's just a couple files out of the whole PR.

File: chromium_src/chrome/browser/flags/android/chrome_feature_list.cc [hunk 4]
```diff
@@ -19,9 +19,9 @@
 #define kForceWebContentsDarkMode kForceWebContentsDarkMode, \
     &brave_rewards::features::kBraveRewards,                 \
     &brave_search_conversion::features::kOmniboxBanner,      \
-    &brave_today::features::kBraveNewsFeature,               \
+    &brave_news::features::kBraveNewsFeature,               \
     &brave_vpn::features::kBraveVPNLinkSubscriptionAndroidUI,\
-    &brave_today::features::kBraveNewsV2Feature,             \
+    &brave_news::features::kBraveNewsV2Feature,             \
```
Comment (iefremov) [inline chromium_src/chrome/browser/flags/android/chrome_feature_list.cc:24]: keeping such lists sorted never hurts

File: chromium_src/chrome/browser/DEPS [hunk 5]
```diff
@@ -13,7 +13,7 @@ include_rules = [
   "+brave/components/brave_search_conversion",
   "+brave/components/brave_shields",
   "+brave/components/brave_sync",
-  "+brave/components/brave_today",
+  "+brave/components/brave_news",
```
Comment (iefremov) [inline chromium_src/chrome/browser/DEPS:16]: pls keep sorted

File: browser/net/brave_network_audit_allowed_lists.h [hunk 6]
```diff
@@ -47,7 +47,7 @@ constexpr const char* kAllowedUrlPrefixes[] = {
     // brave A/B testing
     "https://variations.brave.com/seed",
 
-    // Brave Today (production)
+    // Brave News (production)
     "https://brave-today-cdn.brave.com/",
```
Comment (iefremov) [inline browser/net/brave_network_audit_allowed_lists.h:51]: you would need to add new domains if there are any

**Review by iefremov — APPROVED** [review]
chromium_src LGTM

File: browser/net/brave_network_audit_allowed_lists.h (hunk 6 above)
Comment (fallaciousreasoning) [reply browser/net/brave_network_audit_allowed_lists.h:51]: There aren't any :smile:

**Pull request merged**